src/item_factory.cpp

```diff
@@ -886,6 +900,21 @@ void Item_factory::load_gunmod(JsonObject &jo)
     load_basic_info( jo, new_item_template );
 }
 
+void Item_factory::load( islot_magazine &slot, JsonObject &jo )
+{
+    slot.type = jo.get_string( "ammo_type" );
+    slot.capacity = jo.get_int( "capacity" );
+    slot.reliability = jo.get_int( "reliability" );
+    slot.reload_time = jo.get_int( "reload_time" );
+}
+
+void Item_factory::load_magazine(JsonObject &jo)
+{
+    itype *new_item_template = new itype();
+    load_slot( new_item_template->magazine, jo );
```

Comment:
You should also add a call to `load_slot_optional` at the end of `Item_factory::load_basic_info`, so any item can gain magazine properties.

Actually, do we need a new item type in JSON for this? The magazine data could be loaded from the magazine slot of a GENERIC item in `load_basic_info`. The category can be set from JSON as well.